c++/src/kj/common.h

```diff
@@ -1928,6 +1929,51 @@ class ArrayPtr: public DisallowConstCopyIfNotConst<T> {
     return true;
   }
 
+  inline auto operator<=>(decltype(nullptr)) const { 
+    using ReturnType = decltype(ptr[0] <=> ptr[0]);
+    return size_ == 0 ? (ReturnType) std::strong_ordering::equal
+                      : (ReturnType) std::strong_ordering::greater;
+  }
+
+  inline auto operator<=>(const ArrayPtr& other) const {
+    //Lexographic 3-way comparitor function 
+    //NOTE: Should not be used to erase equality, as does not short circuit on size
```

Comment:
who is this note for?

Reply:
Anyone wanting to naively replace `a == b` with the default from `<=>`.

Reply:
Comments should explain this better now.